manifests/versionlock.pp

```diff
@@ -0,0 +1,36 @@
+# @summary Locks package from updates.
+#
+# @example Sample usage
+#   zypprepo::versionlock { '0:bash-4.1.2-9.sles12.*':
+#     ensure => present,
+#   }
+#
+# @param ensure
+#   Specifies if versionlock should be `present` or `absent`.
+#
+# @note The resource title must use the format
+#   "%{NAME}-%{VERSION}-%{RELEASE}.%{ARCH}".  This can be retrieved via
+#   the command `rpm -q --qf ':%{NAME}-%{VERSION}-%{RELEASE}.%{ARCH}'.
+#   Wildcards may be used within token slots, but must not cover seperators,
+#   e.g., 'b*sh-4.1.2-9.*' covers Bash version 4.1.2, revision 9 on all
+#   architectures.
+#
+define zypprepo::versionlock (
+  Enum['present', 'absent'] $ensure = 'present',
```

Comment:
Same here.
IMO this parameter should be simply removed.
Otherwise it may suggest that it's possible to make sure it's not there.

Reply:
It is possible to ensure that it is not there. Only statements that are defined will be locked and running "zypper al package" will be reset at the next Puppet run.

Reply:
I'm in agreement with @Dan33l here. The proper use here would be to port any existing locks you have into Puppet code or Hieradata and let Puppet manage the entire thing. Puppet doesn't do much good as a tool for managing system state if the stuff it manages can also be touched from other sources as well.